src/action_install.rs

```diff
@@ -228,11 +228,26 @@ pub fn check_tars_and_move(name: &str, dirs: &RuaDirs, archive_whitelist: &Index
 	});
 
 	for (file, file_name) in dir_items {
-		fs::rename(&file.path(), checked_tars_dir.join(file_name)).unwrap_or_else(|e| {
-			panic!(
-				"Failed to move {:?} (build artifact) to {:?}, {}",
-				&file, &checked_tars_dir, e,
-			)
-		});
+		let src = &file.path();
+		let dst = &checked_tars_dir.join(file_name);
+
+		fs::rename(src, dst)
+			.or_else(|err| {
+				const EXDEV: i32 = 18; //Invalid cross-device link
+				if err.raw_os_error() != Some(EXDEV) {
```

Comment:
_(original question was: "should we always fall back to copying?")_

Good point! A cursory glance at `man 3 rename` does not seem to bring up anything, though, and I would like to preserve the original error if possible.  
Hmm, I'll look for other possible cases tomorrow after I wake up.

Reply:
So instead of going to sleep I went through the source code of GNU `mv`... ¯\\\_(ツ)_/¯

`movefile()` calls `do_move()` which, weirdly enough, calls `copy()` with `move_mode = true`.
This attempts to `renameatu()` at first and seems to fall back to copying [ONLY](https://github.com/coreutils/coreutils/blob/9b4bb9d28a6a5f84c407f795d518726fd7902121/src/copy.c#L2466) when `errno == EXDEV`,
erroring out otherwise.
Of note are this [giant disclaimer](https://github.com/coreutils/coreutils/blob/9b4bb9d28a6a5f84c407f795d518726fd7902121/src/copy.c#L2451-L2465) about exotic error codes and this [comment about race conditions](https://github.com/coreutils/coreutils/blob/9b4bb9d28a6a5f84c407f795d518726fd7902121/src/copy.c#L2468-L2477).
I might have _missed_ something, as the code is quite complex.

Things learned: moving files robustly is apparently REALLY complicated.

Fun fact: ZFS can [apparently](https://unix.stackexchange.com/questions/452620/is-mv-atomic-operation-between-two-file-systems/452624#comment822084_452624) `rename()` files across disks as long as they belong to the same zpool. Fancy!

Reply:
Wow, that's some very impressive research!
// Also, that first line made me laugh, all too familiar...:D

I tried to look up if there's some crate that would catch these juicy topics, but there doesn't seem to be any. Apparently, the code's not big enough to create a crate. I wonder if the situation calls for making a QA stackoverflow post though. That doesn't hold the PR or anything, just thinking.

Reply:
Thanks, I felt that `rua` deserved it! :)

It's telling that even `coreutils` folks find the situation uncertain.

It might be common enough that it warrants an addition to `std`. `std::io::ErrorKind::MoveAcrossDevices` and `std::fs::move()`..?